src/sentry/integrations/slack/notifications.py

```diff
@@ -15,10 +17,34 @@
 def send_notification_as_slack(
     notification: ActivityNotification, user: User, context: Mapping[str, Any]
 ) -> None:
-    integrations = Integration.objects.filter(
-        organizations__in=[notification.organization],
-        provider=ExternalProviders.SLACK.name.lower(),
-        status=ObjectStatus.VISIBLE,
-    )
-    for integration in integrations:
-        pass
+    external_actors = ExternalActor.objects.filter(
+        provider=ExternalProviders.SLACK.value,
+        actor=user.actor,
+        organization=notification.organization,
+    ).select_related("integration")
+    for external_actor in external_actors:
+        attachment = [build_notification_attachment(notification)]
+        payload = {
+            "token": external_actor.integration.metadata["access_token"],
```

Comment:
All Slack external actors will have an integrations, but we should be defensive and check to make sure it exists.